interview-questions.md

```diff
@@ -7,62 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Indexs are like old recipe cards, all written and organized on cards in a little file. The index how ever is what place the card or "Apple Pie Recipe" is in the order of cards. While a value is assigned to a variable like const let or var. I have used both of these in my code so far. Indexs have been used when trying to access a specific place in an array or trying to find what is in the [3] place in an array. I have also used values when assigning variables to a string, set of numbers, or another variable that i am trying to changing using a built in method.
 
+Researched answer: Index is a way to pull out a select set of values from an array. The index of a value in an array is that value's location within the array. There is a difference between the value and where the value is stored in an array. Values are values assigned to eather Booleans, Strings, arrays, numbers, etc.
 
 4. What is iteration?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Iteration is the amount of times something will happen or repeat and when it will stop. I have used iteration when creating for loops. In the beginning () there are three steps that need to be done before the loop will work. What number is the loop starting at, the iteration or how long the loop will run for, and how it increments or decrements.
 
+Researched answer: iteration definition "the repetition of a process or utterance." Loops allow programs to perform repetative tasks without repeating themselves such as iterating through an array.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: One person is the driver while the other person is the navigator. The drivers responsibility is to have their hands on whatever they are working on at the time and is basically the one doing the work, while the navigator is checking for mistakes, repeating instructions and providing helpful tips along the way. The navigator is not hands on working. I have had several good pair programming partners and expierences one for example was with Don I was struggling with the terminal so he said I should drive first and he would walk me through it and help me if I ran into problems.
 
+Researched answer: "Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in. The two programmers switch roles frequently." It was created from an agile software development framework named Extreme Programming (XP) aimed to give more improved software.
```

Comment:
I know the medium article you're referring too. Just a reminder you will not have an opportunity to pull on articles in an interview and you'll have to speak on from your own understanding.